gdx/src/com/badlogic/gdx/math/CumulativeDistribution.java

```diff
@@ -27,7 +28,10 @@ public CumulativeValue (T value, float frequency, float interval) {
 	private Array<CumulativeValue> values;
 
 	public CumulativeDistribution () {
-		values = new Array<CumulativeValue>(false, 10, CumulativeValue.class);
+		@SuppressWarnings("unchecked")
+		// It is impossible to create a generic array without reflection (I think)
+		CumulativeValue[] tmp = (CumulativeValue[])ArrayReflection.newInstance(CumulativeValue.class, 10);
+		values = new Array<>(tmp);
```

Comment:
Generics only matter on the LHS. I prefer to omit them from the RHS when possible, except when using `var`. I think it can be:
```
values = new Array(false, 10, CumulativeValue[]::new);
```

Reply:
`values = new Array(false, 10, CumulativeValue[]::new);`
This doesn't work, because `CumulativeValue` has a generic field based on the outer class. And it is impossible (?) to create a generic element array.
Regarding the RHS generic, the java linter issues a warning when omiting it. So to keep the linter output cleaner/seeing actual warnings, I would think keeping the RHS generic is preferable.

Reply:
Nope.
![Doesn't work at all.](https://i.imgur.com/GmBwm2G.png)

Reply:
There's also a "rawtypes" warning issued whenever you omit generics on the RHS. This is textbook Java and has been for almost as long as I have programming it, since I started with Java 6 (before the diamond operator).

Reply:
Try:
```
values = new Array(false, 10, CumulativeDistribution.CumulativeValue[]::new);
```
`rawtypes` warnings are uninteresting.

Reply:
The error can be fixed by making CumulativeValue a static class, with a type parameter for its T value that should just be the same as the CumulativeDistribution's T. I have the changes and can commit them if you want.

Reply:
> The error can be fixed by making CumulativeValue a static class, with a type parameter for its T value that should just be the same as the CumulativeDistribution's T. I have the changes and can commit them if you want.

I implemented that. Since `CumulativeValue` was never exposed outside `CumulativeDistribution`, the change should be non-breaking

Reply:
`CumulativeDistribution.CumulativeValue[]::new` compiled with no other changes (before changing to a static class). Static is better though, as it doesn't make any use of the enclosing instance. CumulativeValue should probably be private.